pkg/swarm/utilities.go

```diff
@@ -0,0 +1,40 @@
+// Copyright 2022 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package swarm
+
+func AddressSliceContains(addrs []Address, a Address) bool {
+	return FindAddressIdx(addrs, a) != -1
+}
+
+func AddressSliceRemove(addrs []Address, a Address) []Address {
+	i := FindAddressIdx(addrs, a)
+	if i == -1 {
+		return addrs
+	}
+
+	return append(addrs[:i], addrs[i+1:]...)
+}
+
+func FindAddressIdx(addrs []Address, a Address) int {
+	for i, v := range addrs {
+		if v.Equal(a) {
+			return i
+		}
+	}
+	return -1
+}
+
+func FindChunkIdxWithAddress(chunks []Chunk, a Address) int {
```

Comment:
I'd recommend renaming it to `IndexOfChunkAddress`.

Reply:
Similarly to `ContainsChunkAddress`, would it be more suitable to name it `IndexOfChunkWithAddress`